src/services/filesManaging.service.ts

```diff
@@ -0,0 +1,69 @@
+import axios from 'axios';
+import * as configApi from './configApi.service';
+import { getToken } from './token.service';
+
+// Perform the upload
+const handleUpload = async (file: File) => {
+  return new Promise<string>((resolve, reject) => {
+    // Split the filename to get the name and type
+    let fileParts = file.name.split('.');
+    let fileName = fileParts[0];
+    let fileType = fileParts[1];
+
+    let key: string = '';
+    axios.post(`${configApi.API_URL}/document/upload`, {
+      fileName: fileName,
+      fileType: fileType
+    }, {
+      headers: { Authorization: `Bearer ${getToken()}` }
+    }).then(response => {
+      const signedUrl = response.data.signedUrl;
+      key = response.data.key;
+
+      console.log("Recieved a signed request " + signedUrl);
+      // Put the fileType in the headers for the upload
+      const options = {
+        headers: {
+          'Content-Type': fileType
+        }
+      };
+      resolve(key);
+
+      axios.put(signedUrl, file, options)
+        .then(result => {
+          console.log("Response from s3")
+        })
+        .catch(error => {
+          alert("ERROR " + JSON.stringify(error));
```

Comment:
It's not safe to print the error to the user. You should only say "An error has occured" for example

Reply:
My bad !